src/adapters/SwivelAdapter.sol

```diff
@@ -7,74 +7,27 @@ import {ISwivel} from 'src/interfaces/ISwivel.sol';
-    uint256 public constant HOLD = 3 days;
-
-    /// @notice address that is allowed to set and withdraw fees, disable principals, etc. It is commonly used in the authorized modifier.
-    address public admin;
-    /// @notice address of the MarketPlace contract, used to access the markets mapping
-    address public marketPlace;
-    /// @notice mapping that determines if a principal has been paused by the admin
-    mapping(uint8 => bool) public paused;
-    /// @notice flag that allows admin to stop all lending and minting across the entire protocol
-    bool public halted;
-
-    /// @notice protocol specific addresses that adapters reference when executing lends
-    /// @dev these addresses are references by an implied enum; adapters hardcode the index for their protocol
-    address[] public protocolRouters;
-
-    /// @notice a mapping that tracks the amount of unswapped premium by market. This underlying is later transferred to the Redeemer during Swivel's redeem call
-    mapping(address => mapping(uint256 => uint256)) public premiums;
-
-    /// @notice this value determines the amount of fees paid on loans
-    uint256 public feenominator;
-    /// @notice represents a point in time where the feenominator may change
-    uint256 public feeChange;
-    /// @notice represents a minimum that the feenominator must exceed
-    uint256 public constant MIN_FEENOMINATOR = 500;
-
-    /// @notice maps underlying tokens to the amount of fees accumulated for that token
-    mapping(address => uint256) public fees;
-    /// @notice maps a token address to a point in time, a hold, after which a withdrawal can be made
-    mapping(address => uint256) public withdrawals;
-
-    // Reantrancy protection
-    uint256 private constant _NOT_ENTERED = 1;
-    uint256 private constant _ENTERED = 2;
-    uint256 private _status = _NOT_ENTERED;
-
-    // Rate limiting protection
-    /// @notice maximum amount of value that can flow through a protocol in a day (in USD)
-    uint256 public maximumValue = 250_000e27;
-    /// @notice maps protocols to how much value, in USD, has flowed through each protocol
-    mapping(uint8 => uint256) public protocolFlow;
-    /// @notice timestamp from which values flowing through protocol has begun
-    mapping(uint8 => uint256) public periodStart;
-    /// @notice estimated price of ether, set by the admin
-    uint256 public etherPrice = 2_500;
-
-    function approve(address[] calldata a) external {
-        address[] memory underlyings = abi.decode('', (address[]));
-
-        for (uint256 i = 0; i < underlyings.length; ) {
-            IERC20(underlyings[i]).approve(
-                protocolRouters[0],
-                type(uint256).max
-            );
-            unchecked {
-                i++;
-            }
+contract SwivelAdapter is IAdapter, Lender {
+    constructor() Lender(address(0), address(0), address(0)) {}
+
+    address public lender = address(0);
+
+    modifier authorizedLender() {
```

Comment:
Any reason why were specifically using `authorizedLender()` instead of our generic `authorized(address)` (with address for the method being the lender)?

Also I honestly forget whether the authorized modifiers use our standard error formatting (uint256,uint256,address, address) as well